### ISSUE TITLE
Use LSTM to Predict Rock-Paper-Scissor 

### ISSUE THREAD

**Issue by gyang1011** [fchollet/keras#5332]
Hello!

I want to create a Neural Network with LSTM to predict opponent's next move in a Rock-Paper-Scissor game. 

So the idea is following, I have encode the inputs as Rock: [1 0 0], Paper: [0 1 0], Scissor: [0 0 1], therefore the dimension is 3 for each input.  The network should take 4 input vectors and produce a output vector as prediction. 

I am still quiet new with Keras and I am not sure if my code for NN generation is correct. Could anyone check it for me? I really appreciate any help that you can provide! 

    #usr/bin/python
    from __future__ import print_function

    from keras.models import Sequential
    from keras.layers import Dense, Activation, Flatten, Dropout
    from keras.layers.recurrent import LSTM
    from keras.optimizers import Adam

    output_dim = 3
    input_dim = 3
    input_length = 4
    batch_size = 20   #use all the data to train in one iteration

    #each input has such strcture
    #Rock: [1 0 0], Paper: [0 1 0], Scissor: [0 0 1]
    #4 inputs (vectors) are sent to the LSTM net and output 1 vector as the prediction

    def createNet(summary=False):
        print("Start Initialzing Neural Network!")
        model = Sequential()
        model.add(LSTM(4,input_dim=input_dim,input_length=input_length,
                return_sequences=True,activation='softmax'))
        model.add(Dropout(0.1))
        model.add(LSTM(4,input_dim=input_dim,input_length=input_length,
                return_sequences=True,activation='softmax'))
        model.add(Dropout(0.1))
        model.add(Flatten())
        model.add(Dense(3,activation='softmax'))

        model.compile(loss='categorical_crossentropy',optimizer='Adam',metrics=['accuracy'])
        if summary:
            print(model.summary())
        return model

    if __name__=='__main__':
        createNet(True)

**Comment by unrealwill** [comment]
Hello, you are on the right track. But there are a few problems with your code.
The last LSTM layer should have a return_sequences=False, and then you don't need to add a Flatten afterwards.
You don't need to specify input_length except for the first layer.
Your LSTM don't have enough cells you need at least 100.
You should probably add some hidden dense layers after the last LSTM.
You can use sparse_categorical_crossentropy
You can use the LSTM dropout input instead.

" If your issue is an implementation question, please ask your question on StackOverflow or join the Keras Slack channel and ask there instead of filing a GitHub issue."

**Comment by gyang1011** [comment]
Thanks a lot!
